estraverse.js

```diff
@@ -390,6 +390,17 @@
     }
 
     Controller.prototype.traverse = function traverse(root, visitor) {
+        function candidateExistInLeaveList(leavelist, candidate) {
```

Comment:
Can't this just be `leavelist.indexOf(candidate) >= 0`?

Reply:
It couldn't. Because there compared element's property `node` with candidate. 

```
leavelist[current].node === candidate
```